vent/sqldb/adapters/postgres_adapter.go

```diff
@@ -185,98 +210,86 @@ func (adapter *PostgresAdapter) UpsertQuery(table types.SQLTable) types.UpsertQu
+
 }
 
 // FindTableQuery returns a query that checks if a table exists
-func (adapter *PostgresAdapter) FindTableQuery(tableName string) string {
-	query := `
-		SELECT EXISTS
-			(
-				SELECT
-					1
-				FROM
-					pg_catalog.pg_class c
-					JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
-				WHERE
-					n.nspname = '%s'
-					AND c.relname = '%s'
-					AND c.relkind = 'r'
-			);`
-
-	return fmt.Sprintf(query, adapter.Schema, tableName)
+func (adapter *PostgresAdapter) FindTableQuery() string {
+	query := "SELECT COUNT(*) found FROM %s.%s WHERE %s = $1;"
+
+	return fmt.Sprintf(query,
+		adapter.Schema, types.SQLDictionaryTableName, //from
+		types.SQLColumnNameTableName) //where
+
 }
 
 // TableDefinitionQuery returns a query with table structure
-func (adapter *PostgresAdapter) TableDefinitionQuery(tableName string) string {
-	return fmt.Sprintf(`
-		WITH dsc AS
-		(
-			SELECT
-				pgd.objsubid,
-				st.schemaname,
-				st.relname
-			FROM
-				pg_catalog.pg_statio_all_tables AS st
-				INNER JOIN pg_catalog.pg_description pgd ON (pgd.objoid = st.relid)
-		)
-		SELECT
-			c.column_name ColumnName,
-			(
-				CASE
-					WHEN c.data_type = 'integer' THEN %v
-					WHEN c.data_type = 'boolean' THEN %v
-					WHEN c.data_type = 'bytea' THEN %v
-					WHEN c.data_type = 'text' THEN %v
-					WHEN c.udt_name = 'timestamp' THEN %v
-					WHEN c.udt_name = 'varchar' THEN %v
-					ELSE 0
-				END
-			) ColumnSQLType,
-			(
-				CASE
-					WHEN is_nullable = 'NO' THEN true
-					ELSE false
-				END
-			) ColumnIsPK,
-			COALESCE(c.character_maximum_length,0) ColumnLength
-		FROM
-			information_schema.columns AS c
-		LEFT OUTER JOIN
-			dsc ON (c.ordinal_position = dsc.objsubid AND c.table_schema = dsc.schemaname AND c.table_name = dsc.relname)
-		WHERE
-			c.table_schema = '%s'
-			AND c.table_name = '%s';`,
-		types.SQLColumnTypeInt,
-		types.SQLColumnTypeBool,
-		types.SQLColumnTypeByteA,
-		types.SQLColumnTypeText,
-		types.SQLColumnTypeTimeStamp,
-		types.SQLColumnTypeVarchar,
-		adapter.Schema,
-		tableName,
-	)
+func (adapter *PostgresAdapter) TableDefinitionQuery() string {
+	query := `
+		SELECT 
+			%s,%s,%s,%s 
+		FROM 
+			%s.%s 
+		WHERE 
+			%s=$1 
+		ORDER BY
+			%s;`
+
+	return fmt.Sprintf(query,
+		types.SQLColumnNameColumnName, types.SQLColumnNameColumnType, //select
+		types.SQLColumnNameColumnLength, types.SQLColumnNamePrimaryKey, //select
+		adapter.Schema, types.SQLDictionaryTableName, //from
+		types.SQLColumnNameTableName,   //where
+		types.SQLColumnNameColumnOrder) //order by
+
 }
 
 // AlterColumnQuery returns a query for adding a new column to a table
-func (adapter *PostgresAdapter) AlterColumnQuery(tableName string, columnName string, sqlColumnType types.SQLColumnType) string {
+func (adapter *PostgresAdapter) AlterColumnQuery(tableName string, columnName string, sqlColumnType types.SQLColumnType, length int, order int) (string, string) {
 	sqlType, _ := adapter.TypeMapping(sqlColumnType)
```

Comment:
`AlterColumnQuery(tableName string, columnName string, sqlColumnType types.SQLColumnType, length int, order int) (string, string)`
should be
`AlterColumnQuery(tableName, columnName string, sqlColumnType types.SQLColumnType, lengtht, order int) (string, string)` instead

Reply:
ready